apps/cookbook/src/app/page/side-nav/side-nav.component.ts

```diff
@@ -1,20 +1,160 @@
-import { Component, OnInit, EventEmitter, Output, Input } from '@angular/core';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  QueryList,
+  ViewChildren,
+} from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+
+import { routes } from '../../showcase/showcase.routes';
+import { navigationItems } from '../header/header.component';
+
+const KEY_DOWN = 'ArrowDown';
+const KEY_UP = 'ArrowUp';
+const HIDDEN_CLASS = 'is-hidden';
+
+interface ISideNavLink {
+  path: string;
+  name: string;
+  active: boolean;
+  hidden: boolean;
+}
 
 @Component({
   selector: 'cookbook-side-nav',
   templateUrl: './side-nav.component.html',
   styleUrls: ['./side-nav.component.scss'],
 })
 export class SideNavComponent implements OnInit {
+  private allShowcaseRoutes: ISideNavLink[];
+  filteredShowcaseRoutes: ISideNavLink[];
+  filter: string = '';
+
   @Output() menuToggle = new EventEmitter<boolean>();
   @Input() isMenuOpen = false;
 
-  constructor() {}
+  constructor(private router: Router) {}
+
+  navigationLinks = navigationItems;
+
+  ngOnInit() {
+    this.mapRoutes();
+    this.filteredShowcaseRoutes = this.allShowcaseRoutes;
+    this.router.events.subscribe((val) => {
+      if (val instanceof NavigationEnd) {
+        if (val.url.indexOf('showcase') === -1) {
+          // reset component links
+          this.filterComponents();
+        }
+      }
+    });
+  }
+
+  private mapRoutes() {
+    const routesWithPath = routes[0].children.filter((r) => r.path);
+    routesWithPath.sort((a, b) => {
+      return a.path < b.path ? -1 : a.path > b.path ? 1 : 0;
+    });
+    this.allShowcaseRoutes = routesWithPath.map((route) => {
+      return {
+        path: `showcase/${route.path}`,
+        name: this.convertKebabToTitleCase(route.path),
+        active: this.router.url.indexOf(route.path) > -1,
+        hidden: false,
+      };
+    });
+  }
+
+  @ViewChildren('componentLink') componentLinks: QueryList<any>;
```

Comment:
Consider strong typing:
```suggestion
  @ViewChildren('componentLink') componentLinks: QueryList<ElementRef<HTMLAnchorElement>>;
```